lib/std/Thread.zig

```diff
@@ -414,6 +414,9 @@ fn callFn(comptime f: anytype, args: anytype) switch (Impl) {
             @call(.{}, f, args) catch |err| {
                 std.debug.print("error: {s}\n", .{@errorName(err)});
                 if (@errorReturnTrace()) |trace| {
+                    if (builtin.zig_backend != .stage1)
+                        trace.index -= 1; // exclude this error-handling block
```

Comment:
In the example you gave, the extra line pointing at `catch` is nice. However, the introduction of this code makes me think this change is not worth it.

Reply:
Fair enough

My main argument for the change is that it can be confusing to understand the "U-turn" taken by an error return trace when handling an error. It also may not be terribly obvious to users that the "parent" error trace is included intentionally in the first place

However, go ahead and close this if you think this isn't the best approach 🙂 We can always circle back later anyway

Reply:
We can also continue the discussion - maybe we can come up with a way to gain this enhancement without a perceived complexity increase in the language

Reply:
Great, let's keep brainstorming 

The main complexity increase I see is for folks writing panic handlers / error trace printers who want to hide the function doing the dumping from the error trace. Is that the increase that jumps out to you?

If so, we might be able to sweep some of that complexity under the rug by adding `std.debug.dumpErrorTrace` or similar